File: dashlane-project/Dashlane/src/main/java/com/dashlane/createaccount/CreateAccountActivity.java
package com.dashlane.createaccount;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.dashlane.R;
import com.dashlane.cryptography.ObfuscatedByteArray;
import com.dashlane.dagger.singleton.SingletonProvider;
import com.dashlane.login.dagger.TrackingId;
import com.dashlane.login.sso.ContactSsoAdministratorDialogFactory;
import com.dashlane.ui.AutoFillDisablerKt;
import com.dashlane.ui.ScreeenshotEnablerKt;
import com.dashlane.ui.activities.DashlaneActivity;
import com.dashlane.ui.endoflife.EndOfLife;

import javax.inject.Inject;

import androidx.annotation.Nullable;
import androidx.lifecycle.LifecycleOwnerKt;
import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class CreateAccountActivity extends DashlaneActivity {

    public static final String EXTRA_PRE_FILLED_EMAIL = "pre_filled_email";
    public static final String EXTRA_SKIP_EMAIL_IF_PRE_FILLED = "skipEmailIfPrefilled";

    @Inject
    CreateAccountContract.DataProvider mDataProvider;

    @Inject
    ContactSsoAdministratorDialogFactory mContactSsoAdministratorDialogFactory;

    @Inject
    @TrackingId
    String trackingId;

    @Inject
    EndOfLife mEndOfLife;

    CreateAccountPresenter mPresenter;

    @Override
    public boolean getRequireUserUnlock() {
        return false;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AutoFillDisablerKt.disableAutoFill(this);
        setContentView(R.layout.activity_login_create_account);
        ScreeenshotEnablerKt.applyScreenshotAllowedFlag(getWindow(), SingletonProvider.getScreenshotPolicy());

        View view = findViewById(R.id.view_login_root);
        CreateAccountViewProxy viewProxy = new CreateAccountViewProxy(view);
        mPresenter = new CreateAccountPresenter(LifecycleOwnerKt.getLifecycleScope(this),
                                                getIntent().getStringExtra(EXTRA_PRE_FILLED_EMAIL),
                                                getIntent().getBooleanExtra(EXTRA_SKIP_EMAIL_IF_PRE_FILLED, false),
                                                mEndOfLife,
                                                mContactSsoAdministratorDialogFactory);
        mPresenter.setView(viewProxy);
        mPresenter.setProvider(mDataProvider);
        ObfuscatedByteArray password = (ObfuscatedByteArray) getLastCustomNonConfigurationInstance();
        mPresenter.setPassword(password);
        mPresenter.onCreate(savedInstanceState);
    }

    @Override
    protected void onStart() {
        super.onStart();
        mPresenter.onStart();
    }

    @Override
    public void onBackPressed() {
        if (!mPresenter.onBackPressed()) {
            super.onBackPressed();
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mPresenter.onDestroy();
    }

    @Override
    public Object onRetainCustomNonConfigurationInstance() {
        return mPresenter.getPassword();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        mPresenter.onSaveInstanceState(outState);
    }

    @SuppressWarnings("DEPRECATION")
    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        mPresenter.onActivityResult(requestCode, resultCode, data);
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/createaccount/CreateAccountActivity.kt
package com.dashlane.createaccount

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.lifecycle.lifecycleScope
import com.dashlane.R
import com.dashlane.cryptography.ObfuscatedByteArray
import com.dashlane.login.dagger.TrackingId
import com.dashlane.login.sso.ContactSsoAdministratorDialogFactory
import com.dashlane.ui.activities.DashlaneActivity
import com.dashlane.ui.disableAutoFill
import com.dashlane.ui.endoflife.EndOfLife
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class CreateAccountActivity : DashlaneActivity() {

    @Inject
    lateinit var dataProvider: CreateAccountContract.DataProvider

    @Inject
    lateinit var contactSsoAdministratorDialogFactory: ContactSsoAdministratorDialogFactory

    @Inject
    @TrackingId
    lateinit var trackingId: String

    @Inject
    lateinit var endOfLife: EndOfLife

    lateinit var presenter: CreateAccountPresenter

    override var requireUserUnlock: Boolean
        get() = false
        set(requireUserUnlock) {
            super.requireUserUnlock = requireUserUnlock
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.disableAutoFill()
        setContentView(R.layout.activity_login_create_account)
        val view = findViewById<View>(R.id.view_login_root)
        val viewProxy = CreateAccountViewProxy(view)

        presenter = CreateAccountPresenter(
            coroutineScope = this.lifecycleScope,
            preFilledEmail = intent.getStringExtra(EXTRA_PRE_FILLED_EMAIL),
            skipEmailIfPrefilled = intent.getBooleanExtra(EXTRA_SKIP_EMAIL_IF_PRE_FILLED, false),
            endOfLife = endOfLife,
            contactSsoAdministratorDialogFactory = contactSsoAdministratorDialogFactory
        ).apply {
            setView(viewProxy)
            setProvider(dataProvider)
            password = lastCustomNonConfigurationInstance as ObfuscatedByteArray?
            onCreate(savedInstanceState)
        }
    }

    override fun onStart() {
        super.onStart()
        presenter.onStart()
    }

    override fun onBackPressed() {
        if (!presenter.onBackPressed()) {
            super.onBackPressed()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        presenter.onDestroy()
    }

    override fun onRetainCustomNonConfigurationInstance(): Any? {
        return presenter.password
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        presenter.onSaveInstanceState(outState)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        presenter.onActivityResult(requestCode, resultCode, data)
    }

    companion object {
        const val EXTRA_PRE_FILLED_EMAIL = "pre_filled_email"
        const val EXTRA_SKIP_EMAIL_IF_PRE_FILLED = "skipEmailIfPrefilled"
    }
}